library_information_validator.py

```diff
@@ -0,0 +1,495 @@
+                if(current_cell_value_pool == pool):
+                    current_pool_rows.append(nrow_nr)
+            validator = Validator(self.library_sheet, None)
+
+            result_index, sindex, warning_index_mix, \
+            warning_length_comp = validator.select_index(current_pool_rows, pool)
+
+            warning_low_div, warning_index_length,\
+            warning_index_balance = validator.validate_index(result_index, pool, sindex)
+
+            pool_warning = pool, [warning_low_div, warning_index_length, \
+            warning_index_balance, warnings_cycle[i], warning_index_mix, warning_length_comp]
+
+            pool_warnings.append(pool_warning)
+            current_pool_rows =[]
+            i += 1
+
+        # summarise warnings and return
+        pools_with_warnings = []
+        for warning in pool_warnings:
+            sums_warnings = sum(warning[1])
+            if(sums_warnings > 0):
+                pools_with_warnings.append(warning[0])
+        return(len(pools_with_warnings), len(poolIDs))
+
+class Validator(object):
+    # Initializer / Instance attributes
+    def __init__(self, access_sample_info_sheet, molarityID):
+        self.access_sample_info_sheet = access_sample_info_sheet
+        self.molarityID = molarityID
+
+    # instance methods
+    def validate_numeric(self):
+        """Checks whether value is numeric or not."""
+        warnings_numeric = 0
+        for checkNumbers in [self.molarityID]:
+            if not isinstance(self.access_sample_info_sheet[checkNumbers].value, numbers.Number):
+                try:
+                    float(self.access_sample_info_sheet[checkNumbers].value.replace(",", "."))
+                    logger.error(
+                        'Cell {} with value \"{}\" is not numeric due to decimal point/comma clash.'\
+                        .format(self.access_sample_info_sheet[checkNumbers].coordinate, self.access_sample_info_sheet[checkNumbers].value)
+                        )
+                    warnings_numeric += 1
+                    return False
+                except ValueError:
+                    logger.error(
+                    'Cell {} with value {} is \"'.format(self.access_sample_info_sheet[checkNumbers].coordinate)+ self.access_sample_info_sheet[checkNumbers].value + '\" is not numeric'
+                    )
+                    warnings_numeric += 1
+                except TypeError:
+                    if self.access_sample_info_sheet[checkNumbers].value is None:
+                        logger.error(
+                        'Cell {} is numeric but empty'.format(self.access_sample_info_sheet[checkNumbers].coordinate)
+                        )
+                        warnings_numeric += 1
+                        return False
+                    else:
+                        raise
+            return(False, warnings_numeric)
+
+    # currently not used
+    #def validate_mol(self, min_mol, max_mol):
```

Comment:
How could this part of the code be used?

Reply:
This is for future implementation of a check on pool molarity. I am waiting for Mattias to do some stuff, so Johannes (or me) can create a database, which is required to know which application requires what molarity.